pkg/virt-handler/vm.go

```diff
@@ -152,6 +153,13 @@ type VirtualMachineController struct {
 	podIsolationDetector     isolation.PodIsolationDetector
 	containerDiskMounter     *container_disk.Mounter
 	clusterConfig            *virtconfig.ClusterConfig
+
+	// records if pod network phase1 has completed
+	// phase1 involves cycling an entire posix thread
+	// so for performance, knowing phase1 is complete
+	// prevents cycling an unncessary posix thread.
+	phase1NetworkSetupCache     map[types.UID]int
```

Comment:
There is a thread-safe sync.Map, turns out it's optimized for other use cases.
https://tip.golang.org/pkg/sync/#Map